packages/locale-loader/lib/parseLocaleFile/index.js

```diff
@@ -0,0 +1,40 @@
+import { parse } from 'babylon';
+import { find, forEach } from 'ramda';
+import generate from 'babel-generator';
+import extractAnnotations from '../extractAnnotations';
+
+/* eslint { no-eval: 0 } */
+export default function parseLocaleFile(rawContent) {
+  const data = new Map();
+  const { content, annotations } = extractAnnotations(rawContent);
+  const ast = parse(content, { sourceType: 'module' });
+
+  const defaultExport = find(
+    item => item.type === 'ExportDefaultDeclaration',
+    ast.program.body,
+  );
+  if (defaultExport && defaultExport.declaration.type === 'ObjectExpression') {
+    forEach((prop) => {
+      const key = prop.key.type === 'MemberExpression' ?
+        `[${generate(prop.key).code}]` :
+        generate(prop.key).code;
+      const value = eval(generate(prop.value).code);
```

Comment:
Each node of a AST contains a `start` and `end` fields that indicate the location of source code, which can be used to extract code snippets.